### PR TITLE
fix: addressing sandwich attack fix in the deposit function

### PR DESCRIPTION
**Pull request by kupermind**
- Addressing sandwich attack fix in the `deposit()` function as specified in #13.

### CODE REVIEW

File: lockbox2/programs/liquidity_lockbox/src/lib.rs [hunk 1]
```diff
@@ -266,7 +240,7 @@ pub mod liquidity_lockbox {
       cpi_accounts_modify_liquidity,
       signer_seeds
     );
-    whirlpool::cpi::increase_liquidity(cpi_ctx_modify_liquidity, liquidity_amount, delta_a, delta_b)?;
+    whirlpool::cpi::increase_liquidity(cpi_ctx_modify_liquidity, liquidity_amount as u128, token_max_a, token_max_b)?;
```
Comment (mariapiamo) [inline lockbox2/programs/liquidity_lockbox/src/lib.rs:243]: what about the approval?

**Review by mariapiamo — APPROVED** [review]

File: lockbox2/programs/liquidity_lockbox/src/lib.rs (hunk 1 above)
Comment (kupermind) [reply lockbox2/programs/liquidity_lockbox/src/lib.rs:243]: It's in the code below :)

Comment (kupermind) [reply lockbox2/programs/liquidity_lockbox/src/lib.rs:243]: We approve the max provided amounts, then approve back to zero

Comment (mariapiamo) [reply lockbox2/programs/liquidity_lockbox/src/lib.rs:243]: Oh sorry! the approval is above for ` token_max_a` ` token_max_b` !!

**Review by mariapiamo — APPROVED** [review]

File: lockbox2/programs/liquidity_lockbox/src/lib.rs (hunk 1 above)
Comment (77ph) [reply lockbox2/programs/liquidity_lockbox/src/lib.rs:243]: Fix for .unwrap_or_else(|| panic!("Liquidity overflow")) in another PR?

Comment (kupermind) [reply lockbox2/programs/liquidity_lockbox/src/lib.rs:243]: This will be done in a next PR

Comment (77ph) [reply lockbox2/programs/liquidity_lockbox/src/lib.rs:243]: LGTM

**Review by 77ph — APPROVED** [review]

File: lockbox2/programs/liquidity_lockbox/src/lib.rs [hunk 2]
```diff
@@ -121,9 +118,11 @@ pub mod liquidity_lockbox {
   /// Deposits an NFT position under the Lockbox management and gets bridged tokens minted in return.
   ///
   /// ### Parameters
+  /// - `liquidity_amount` - Requested liquidity amount.
   /// - `token_max_a` - Max amount of SOL token to be added for liquidity.
   /// - `token_max_b` - Max amount of OLAS token to be added for liquidity.
   pub fn deposit(ctx: Context<DepositPositionForLiquidity>,
+    liquidity_amount: u64,
```
Comment (kupermind) [inline lockbox2/programs/liquidity_lockbox/src/lib.rs:125]: Adding a `liquidity_amount` parameter.

File: lockbox2/programs/liquidity_lockbox/src/lib.rs [hunk 3]
```diff
@@ -185,35 +184,10 @@ pub mod liquidity_lockbox {
...
-      return Err(ErrorCode::DeltaAmountOverflow.into());
-    }
-
-    // Check that the liquidity is within uint64 bounds
-    if liquidity_amount > std::u64::MAX as u128 {
-      return Err(ErrorCode::LiquidityOverflow.into());
-    }
-
-    let position_liquidity = liquidity_amount as u64;
-
```
Comment (kupermind) [inline lockbox2/programs/liquidity_lockbox/src/lib.rs:212]: This is obsolete as the liquidity amount is already provided.

File: lockbox2/programs/liquidity_lockbox/src/lib.rs [hunk 4]
```diff
@@ -226,7 +200,7 @@ pub mod liquidity_lockbox {
           authority: ctx.accounts.signer.to_account_info(),
         },
       ),
-      delta_a,
+      token_max_a,
```
Comment (kupermind) [inline lockbox2/programs/liquidity_lockbox/src/lib.rs:203]: We now approve for max amount of tokens.

File: lockbox2/programs/liquidity_lockbox/src/lib.rs [hunk 5]
```diff
@@ -287,12 +261,36 @@ pub mod liquidity_lockbox {
...
+      CpiContext::new(
+        ctx.accounts.token_program.to_account_info(),
+        Approve {
+          to: ctx.accounts.token_owner_account_b.to_account_info(),
+          delegate: ctx.accounts.lockbox.to_account_info(),
+          authority: ctx.accounts.signer.to_account_info(),
+        },
+      ),
+      0,
+    )?;
```
Comment (kupermind) [inline lockbox2/programs/liquidity_lockbox/src/lib.rs:288]: Revoke the approval of unused tokens.

File: lockbox2/programs/liquidity_lockbox/src/lib.rs [hunk 6]
```diff
@@ -455,34 +453,6 @@ pub mod liquidity_lockbox {
...
 
-// https://github.com/orca-so/whirlpools/blob/main/sdk/src/quotes/public/increase-liquidity-quote.ts#L147
-// https://github.com/orca-so/whirlpools/blob/537306c096bcbbf9cb8d5cff337c989dcdd999b4/sdk/src/utils/position-util.ts#L69
-/// Gets liquidity from token A parameters.
-///
-/// ### Parameters
-/// - `amount` - Token A amount.
-/// - `sqrt_price_lower_x64` - Lower sqrt price.
-/// - `sqrt_price_upper_x64` - Upper sqrt price.
-fn get_liquidity_from_token_a(amount: u128, sqrt_price_lower_x64: u128, sqrt_price_upper_x64: u128 ) -> Result<u128> {
```
Comment (kupermind) [inline lockbox2/programs/liquidity_lockbox/src/lib.rs:466]: This becomes obsolete as there's no need to calculate a liquidity here (already provided).

**Pull request merged**